curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/66f59e667892883f47abe6d5.md

```diff
@@ -0,0 +1,162 @@
+---
+id: 66f59e667892883f47abe6d5
+title: Step 35
+challengeType: 0
+dashedName: step-35
+---
+
+# --description--
+
+Your `forEach` loop is fully functional, however we can use an alternate syntax here called an 
+<dfn>arrow function expression</dfn>. This kind of function expression does not need curly braces and 
+is sometimes recommended for shorter functions. 
+
+Here is an example of arrow function syntax. 
+
+```js
+const greetings = name => `Hello, ${name}!`;
+```
+
+Update the callback in your `forEach` to be an arrow function. 
+
+
+# --hints--
+
+Your function should still increment the `counts` variable properly. 
+
+```js
+const expected = {'2': 1, '4': 2, '5': 1};
+assert.deepEqual(getMode([4, 4, 2, 5]), expected)
+```
+
+Your function should use an `arrow function expression`. 
+
+```js
+assert.match(getMode.toString(), /counts\[el\]\s*=\s*\(counts\[el\]\s*\|\|\s*0\)\s*\+\s*1/);
```

Comment:
Just a thought... We already test the function implementation by asserting the returned value is correct. So rather than a regular expression against the function body, should we regex this bit:

```
array.forEach(el =>
```

And assert it is not followed by a `{`?

Reply:
That code also isn't the only valid expression. You could use a ternary.

So yes, we shouldn't force that single expression as the only valid solution.

Reply:
That'd probably be pretty reasonable.